hubspot/plugin.py

```diff
@@ -454,57 +454,95 @@ class HubSpotSearchCondition:
         "LTE",
         "GT",
         "GTE",
+        "HAS_PROPERTY",
+        "NOT_HAS_PROPERTY",
     ]
     value: HubSpotPropertyValue
 
 
-@purpose("Search contacts.")
+@dataclass
+class AndGroup:
+    """A group of conditions that must ALL be true (AND logic). Maximum of 6 conditions"""
+
+    conditions: List[HubSpotSearchCondition]
+
+
+@dataclass
+class SearchQuery:
+    """
+    Top-level search query that combines multiple AND groups with OR logic.
+    Any ONE of the AndGroups must match for a result to be returned.
+
+    Limitations:
+    - Maximum of 5 or_groups.
+    - Maximum 18 total conditions across all groups.
+    """
+
+    or_groups: List[AndGroup]
+
+
+def _convert_and_groups_to_filter_groups(
+    and_groups: List[AndGroup], schema: _HubSpotPropertiesSchema
+) -> List[dict[str, Any]]:
```

Comment:
We could probably used TypedDict to represent the structure of Any here since I think we're in control of all the keys and types